_nx_parallel/config.py

```diff
@@ -0,0 +1,25 @@
+from networkx.utils.configs import Config
+from typing import Union
+from dataclasses import dataclass, field
+
+__all__ = [
+    "ParallelConfig",
+    "_configs",
```

Comment:
When do we choose to use the plural name `configs` and when do we use `config`? It seems like most of the code here and in networkx we use the singular (except for the module name in networkx which kind of makes sense because that code allows many different config to co-exist and the module name does need to be different from local names).  Should this be `_config`?  

Also, we don't usually include private names in `__all__`.  Should this be a public name instead of a private name?  It looks to me that this is only used to set the default config. So I would suggest keeping it private, naming it `_default_config`, and removing it from `__all__`. 

I'm actually a little worried that `ParallelConfig` is a public class. Do we want users to be creating their own config instances? I guess I don't understand the config paradigm fully.